velox/docs/functions/presto/window.rst

```diff
@@ -132,17 +132,19 @@ returned. It is an error for the offset to be zero or negative.
 
 .. function:: lag(x[, offset [, default_value]]) -> [same as input]
 
-Returns the value at ``offset`` rows before the current row in the partition. If
-there is no such row, the ``default_value`` is returned, or if it is not
-specified ``null`` is returned. Offsets start at ``0``, which is the current
-row. The default ``offset`` is ``1``.
+Returns the value at ``offset`` rows before the current row in the window partition.
```

Comment:
I'm seeing that this PR changes the behavior in case when IGNORE NULLS is specific. Would you update this documentation to clarify what happens when IGNORE NULLS is on?

Reply:
by IGNORE NULLS do you mean the `if constexpr (ignoreNulls)` branch in `setRowNumbers` function, I have not touch that part in this PR

Reply:
@aditi-pandit Aditi, it looks like we might be missing documentation for IGNORE NULLS clause: https://facebookincubator.github.io/velox/functions/presto/window.html

Reply:
I'd suggest adding document for IGNORE NULL in a separate PR, seems many functions has `IGNORE NULLS` support(see [1]), so it might be better to describe IGNORE NULLS as a dedicated section.

[1]. https://duckdb.org/docs/sql/window_functions.html

Reply:
Is the behavior of IGNORE NULLS  consistent across all functions? I assume lead / lag have their own semantics, no? Let's clarify what these are and make sure the code work properly.

Reply:
Added some doc for IGNORE NULLs for lead/lag.

Reply:
> I'd suggest adding document for IGNORE NULL in a separate PR, seems many functions has `IGNORE NULLS` support(see [1]), so it might be better to describe IGNORE NULLS as a dedicated section.
> 
> [1]. https://duckdb.org/docs/sql/window_functions.html

@xumingming , @mbasmanova : I'll add doc for IGNORE NULLs in a later PR.

Reply:
@aditi-pandit Thank you, Aditi. Can you help clarify IGNORE NULLS behavior for lead/lag so we can make sure the implementation is 100% correct?

Reply:
The meaning of IGNORE NULLs in lead/lag is as follows:

lead/lag need to get the column value of a row 'offset' rows away for the result. For IGNORE NULLs, we have to find the column value which is non-null 'offset' rows away. 

So the question for this PR is the code in this [comment ](https://github.com/facebookincubator/velox/pull/7254/files#r1376350944)... If there aren't non-null offset number of rows, then what should the result be ? As per Presto java it is the default value (if specified).

So this code is correct imo. 

But of course, its better to add a test. I think we can just add one more column in [ignoreNulls test](https://github.com/facebookincubator/velox/blob/main/velox/functions/prestosql/window/tests/LeadLagTest.cpp#L391) that has all null values only. By using that as a parameter for lead/lag should return default value for each row.

Reply:
Done, added all null values test.